### PR TITLE
feat: support groupby on growing segment(#30307)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #30307

### CODE REVIEW

File: internal/core/src/query/GroupByOperator.h [hunk 1]
```diff
@@ -19,19 +19,53 @@
...
+            segment.get_insert_record().get_field_data<T>(fieldId);
+    }
+
+    GrowingDataGetter(const GrowingDataGetter<T>& other)
+        : growing_raw_data_(other.growing_raw_data_) {
+    }
+
+    T
+    Get(int64_t idx) const {
+        const T& val_ref = growing_raw_data_->operator[](idx);
```
Comment (yah01) [inline internal/core/src/query/GroupByOperator.h:53]: why not simply return this

**Review by yah01 — APPROVED** [review]
/lgtm

File: internal/core/src/query/GroupByOperator.h (hunk 1 above)
Comment (MrPresent-Han) [reply internal/core/src/query/GroupByOperator.h:53]: I will fix this in the next pr

**Comment by MrPresent-Han** [comment]
rerun ut

**Pull request closed without merging**